irpp_mensuel_salarie > 0 pour ouvrière à 427 TND par mois

Afin de tester `openjibaya`, je tente de construire un test utilisant les variables dont a besoin `openjibaya` sur les données issues de la fiche de paie ouvrière.

Je m'attends à ce que ce test YAML passe :
```
- name: OpenJibaya sur cas ouvrière
  reforms: 
    - openfisca_tunisia.reforms.de_net_a_salaire_de_base.de_net_a_salaire_de_base
  period: 2016-02
  absolute_error_margin: 1
  input:
    salaire_net_a_payer: 427.040
    primes: 6.080 + 47.500
    
  output:
    irpp_mensuel_salarie: 0 # déduit de ouvriere_2016_02.yaml
    salaire_imposable: 427.040
    
    cotisations_salarie: -43.164 # cnss
    salaire_de_base: 416.624
    
    assiette_cotisations_sociales: 470.204 # brut

    # salaire_super_brut: 427.040 + 0 + 43.164 + 6.080 + 47.500 + cotisations_employeur
```
Les variables qui m'intéressent pour `openjibaya` sont donc les outputs.
Mais le test ne passe pas. J'obtiens différents KO en commençant par : 
```
AssertionError: In test 'OpenJibaya sur cas ouvrière', in file '/Users/sch/Dev/github/tunisie/openfisca-tunisia/tests/reforms/test_salaire.yaml', irpp_mensuel_salarie@2016-02: [-45.468124] differs from 0.0 with an absolute margin [45.468124] > 1
```

Où est l'erreur ?
Serait-ce aussi lié au `regime_securite_sociale` de l'ouvrière ? 🤔

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+### 0.23.0 - [#85](https://github.com/openfisca/openfisca-tunisia/pull/85)
+
+* Improve `reforms/de_net_a_salaire_de_base.py`
+  * Adapt with`ouvriere_2016_02.yaml` use case
+* Add `reforms/de_net_a_salaire_de_base/ouvriere_2016_02.yaml` test
+
 ### 0.22.1 - [#78](https://github.com/openfisca/openfisca-tunisia/pull/78)
 
 * Add some other missing units to `parameters`

--- a/openfisca_tunisia/model/prelevements_obligatoires/cotisations_sociales.py
+++ b/openfisca_tunisia/model/prelevements_obligatoires/cotisations_sociales.py
@@ -35,7 +35,7 @@ def compute_cotisation(individu, period, cotisation_type = None, bareme_name = N
     baremes_by_regime = parameters(period.start).cotisations_sociales
     cotisation = zeros(len(assiette_cotisations_sociales))
     types_regime_securite_sociale = regime_securite_sociale.possible_values
-    
+
     for regime in types_regime_securite_sociale:
         if 'cotisations_{}'.format(cotisation_type) not in baremes_by_regime[regime.name]._children:
             continue
@@ -389,8 +389,8 @@ class salaire_imposable(Variable):
 
     def formula(individu, period):
         return (
-            individu('assiette_cotisations_sociales', period) +
-            individu('cotisations_salarie', period)
+            individu('assiette_cotisations_sociales', period)
+            + individu('cotisations_salarie', period)
             )
 
 
@@ -403,8 +403,8 @@ class salaire_net_a_payer(Variable):
 
     def formula(individu, period):
         return (
-            individu('salaire_imposable', period) +
-            individu('irpp_mensuel_salarie', period)
+            individu('salaire_imposable', period)
+            + individu('irpp_mensuel_salarie', period)
             )
 
 
@@ -417,9 +417,9 @@ class salaire_super_brut(Variable):
 
     def formula(individu, period):
         return (
-            individu('salaire_de_base', period = period) +
-            individu('primes', period = period) -
-            individu('cotisations_employeur', period = period)  # Cotisations employeur sont négatives
+            individu('salaire_de_base', period = period)
+            + individu('primes', period = period)
+            - individu('cotisations_employeur', period = period)  # Cotisations employeur sont négatives
             )
 
 

--- a/openfisca_tunisia/reforms/de_net_a_salaire_de_base.py
+++ b/openfisca_tunisia/reforms/de_net_a_salaire_de_base.py
@@ -11,10 +11,9 @@ except ImportError:
     fsolve = None
 
 
-def calculate_net_from(salaire_de_base, primes, individu, period, requested_variable_names):
+def calculate_net_from(salaire_de_base, individu, period, requested_variable_names):
     # We're not wanting to calculate salaire_imposable again, but instead manually set it as an input variable
     individu.get_holder('salaire_de_base').put_in_cache(salaire_de_base, period)
-    individu.get_holder('primes').put_in_cache(primes, period)
 
     # Work in isolation
     temp_simulation = individu.simulation.clone()
@@ -42,10 +41,7 @@ class salaire_de_base(Variable):
     def formula(individu, period):
         # Calcule le salaire de base à partir du salaire net par inversion numérique.
         net = individu.get_holder('salaire_net_a_payer').get_array(period)
-        primes = individu('primes', period)
 
-        if primes is None:
-            primes = individu.empty_array()
         if net is None:
             return individu.empty_array()
 
@@ -64,7 +60,7 @@ class salaire_de_base(Variable):
 
         def solve_func(net):
             def innerfunc(essai_salaire_de_base):
-                return calculate_net_from(essai_salaire_de_base, primes, individu, period, requested_variable_names) - net
+                return calculate_net_from(essai_salaire_de_base, individu, period, requested_variable_names) - net
             return innerfunc
 
         salaire_de_base_calcule = \

--- a/openfisca_tunisia/reforms/de_net_a_salaire_de_base.py
+++ b/openfisca_tunisia/reforms/de_net_a_salaire_de_base.py
@@ -3,7 +3,7 @@
 from __future__ import division
 
 from openfisca_tunisia.model.base import *
-from openfisca_tunisia import entities
+
 
 try:
     from scipy.optimize import fsolve
@@ -11,9 +11,10 @@ except ImportError:
     fsolve = None
 
 
-def calculate_net_from(salaire_de_base, individu, period, requested_variable_names):
+def calculate_net_from(salaire_de_base, primes, individu, period, requested_variable_names):
     # We're not wanting to calculate salaire_imposable again, but instead manually set it as an input variable
     individu.get_holder('salaire_de_base').put_in_cache(salaire_de_base, period)
+    individu.get_holder('primes').put_in_cache(primes, period)
 
     # Work in isolation
     temp_simulation = individu.simulation.clone()
@@ -34,13 +35,17 @@ def calculate_net_from(salaire_de_base, individu, period, requested_variable_nam
 class salaire_de_base(Variable):
     value_type = float
     entity = Individu
-    label = u"Salaire imposable"
+    label = u"Salaire de base"
     definition_period = MONTH
     set_input = set_input_divide_by_period
 
     def formula(individu, period):
         # Calcule le salaire de base à partir du salaire net par inversion numérique.
         net = individu.get_holder('salaire_net_a_payer').get_array(period)
+        primes = individu('primes', period)
+
+        if primes is None:
+            primes = individu.empty_array()
         if net is None:
             return individu.empty_array()
 
@@ -58,15 +63,15 @@ class salaire_de_base(Variable):
         simulation.requested_periods_by_variable_name = dict()
 
         def solve_func(net):
-            def innerfunc(essai):
-                return calculate_net_from(essai, individu, period, requested_variable_names) - net
+            def innerfunc(essai_salaire_de_base):
+                return calculate_net_from(essai_salaire_de_base, primes, individu, period, requested_variable_names) - net
             return innerfunc
 
         salaire_de_base_calcule = \
             fsolve(
                 solve_func(net),
-                net * 1.4,  # first guess
-                xtol = 1 / 100,  # précision
+                net * 1,  # first guess
+                xtol = 1 / 1000,  # précision au millime
                 )
 
         return salaire_de_base_calcule

--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ doc_lines = __doc__.split('\n')
 
 setup(
     name = 'OpenFisca-Tunisia',
-    version = '0.22.1',
+    version = '0.23.0',
     author = 'OpenFisca Team',
     author_email = 'contact@openfisca.org',
     classifiers = [classifier for classifier in classifiers.split('\n') if classifier],

--- a/tests/fiches_de_paie/ouvriere_2016_02.yaml
+++ b/tests/fiches_de_paie/ouvriere_2016_02.yaml
@@ -9,3 +9,16 @@
     cotisations_salarie: -43.164
     salaire_imposable: 427.040
     salaire_net_a_payer: 427.040
+
+- name: Ouvrière février 2016 imposée mais avec même salaire net que la précédente
+  period: 2016-02
+  absolute_error_margin: 0.5
+  input:
+    salaire_de_base: 466.688
+    primes: 6.080 + 47.500
+
+  output:
+    salaire_net_a_payer: 427.040
+    assiette_cotisations_sociales: 520.268
+    cotisations_salarie: -47.761
+    salaire_imposable: 472.507

--- a/tests/reforms/de_net_a_salaire_de_base/ouvriere_2016_02.yaml
+++ b/tests/reforms/de_net_a_salaire_de_base/ouvriere_2016_02.yaml
@@ -1,0 +1,18 @@
+- name: OpenJibaya sur cas ouvrière
+  reforms:
+    - openfisca_tunisia.reforms.de_net_a_salaire_de_base.de_net_a_salaire_de_base
+  period: 2016-02
+  absolute_error_margin: 1
+  input:
+    salaire_net_a_payer: 427.040
+    primes: 6.080 + 47.500
+
+  output:
+    primes: 6.080 + 47.500
+    salaire_de_base: 416.624
+    assiette_cotisations_sociales: 470.204 # brut
+    salaire_de_base: 416.624
+    salaire_imposable: 427.040
+    cotisations_salarie: -43.164 # cnss
+    irpp_mensuel_salarie: 0 # déduit de ouvriere_2016_02.yaml
+    # salaire_super_brut: 427.040 + 0 + 43.164 + 6.080 + 47.500 + cotisations_employeur

--- a/tests/reforms/de_net_a_salaire_de_base/ouvriere_2016_02.yaml
+++ b/tests/reforms/de_net_a_salaire_de_base/ouvriere_2016_02.yaml
@@ -6,7 +6,6 @@
   input:
     salaire_net_a_payer: 427.040
     primes: 6.080 + 47.500
-
   output:
     primes: 6.080 + 47.500
     salaire_de_base: 416.624
@@ -15,4 +14,3 @@
     salaire_imposable: 427.040
     cotisations_salarie: -43.164 # cnss
     irpp_mensuel_salarie: 0 # déduit de ouvriere_2016_02.yaml
-    # salaire_super_brut: 427.040 + 0 + 43.164 + 6.080 + 47.500 + cotisations_employeur

